lib/takeaway.rb

```diff
@@ -0,0 +1,90 @@
+class Takeaway
+
+  CHECKOUT_ERROR = "Total cost entered does not match the sum of your order!"
+
+  attr_reader :menu
+
+  def initialize(menu = Menu.new)
+    @basket = Hash.new(0)
+    @menu = menu
+  end
+
+  def read_menu
+    @menu.dishes.each { |item, cost| print "#{item}: £#{cost}, " }
+  end
+
+  def order(item, quantity=1)
+    if @menu.dishes[item].nil?
+      "#{item} does not exist!"
+    else
+      @basket[item] += quantity
+      "#{quantity}x #{item}(s) added to your basket."
+    end
+  end
+
+  def basket_summary
+    @basket.each do |item, quantity|
+      print "#{item} x#{quantity} = £#{(@menu.dishes[item]*quantity).round(2)}, "
```

Comment:
Metrics/LineLength: Line is too long. [81/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)